vmicore/src/lib/vmi/InterruptGuard.cpp

```diff
@@ -23,7 +23,7 @@ namespace VmiCore
           logger(logging->newNamedLogger(loggerName)),
           targetVA(targetVA),
           targetGFN(targetGFN),
-          shadowPage(PagingDefinitions::pageSizeInBytes + 16),
+          shadowPage(PagingDefinitions::pageSizeInBytes),
```

Comment:
Please allocate the final size right away. This could lead to expensive reallocations otherwise.